### PR TITLE
Improve the workflow to prevent connection when the solution is unloa…

### LINKED ISSUE #347: "Connect" option is available when no solution is loaded
In team explorer, it is possible to connect to SQ with no loaded solutions. When done, the error message appears on top of team explorer. The messages is subtly wrong:
"This solution contains no supported project language (C#, VB.Net). Please create at least one project of the supported language and then try to reconnect."

Ideally, we wouldn't show the connect option at all, or display a message like "no loaded solution" when clicked.

Related Information:
    SonarLint for VS Version: Milestone 3.6
    Visual Studio Version: 2015u3

### LINKED ISSUE #395: Binding the solution to SonarQube before all projects have finished to load result in weird error
Currently, if the binding process finishes before all the projects of the solution are loaded then the binding fails with some unclear message and leave the solution in an unstable state (doesn't undo but cannot finish properly the binding). From this point it is no longer possible to bind.

### PR DESCRIPTION
**Pull request by Evangelink**
…ded or not fully loaded

Fix #347 and fix #395

### CODE REVIEW

File: src/Integration/MefServices/IActiveSolutionTracker.cs [hunk 1]
```diff
@@ -28,5 +28,7 @@ internal interface IActiveSolutionTracker
         /// The active solution has changed (either opened or closed).
         /// </summary>
         event EventHandler ActiveSolutionChanged;
```
Comment (Evangelink) [inline src/Integration/MefServices/IActiveSolutionTracker.cs]: As the class needs to be stateful I have decided not to put any specific event args to the event.

File: src/Integration/MefServices/ActiveSolutionTracker.cs [hunk 2]
```diff
@@ -86,8 +89,7 @@ int IVsSolutionEvents.OnBeforeUnloadProject(IVsHierarchy pRealHierarchy, IVsHier
 
         int IVsSolutionEvents.OnAfterOpenSolution(object pUnkReserved, int fNewSolution)
```
Comment (Evangelink) [inline src/Integration/MefServices/ActiveSolutionTracker.cs]: This event was triggered when opening the solution not when the solution was fully loaded.

File: src/Integration/Integration.csproj [hunk 3]
```diff
@@ -120,6 +120,11 @@
       <HintPath>..\..\packages\Microsoft.VisualStudio.Shell.Interop.7.10.6071\lib\Microsoft.VisualStudio.Shell.Interop.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Microsoft.VisualStudio.Shell.Interop.10.0, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <EmbedInteropTypes>True</EmbedInteropTypes>
```
Comment (duncanp-sonar) [inline src/Integration/Integration.csproj]: Out of interest, did VS set _EmbedInteropTypes_ to true?

File: src/Integration.UnitTests/Connection/ConnectControllerTests.cs [hunk 4]
```diff
@@ -93,6 +96,7 @@ public void ConnectionController_ConnectCommand_Status()
         {
             // Arrange
             var testSubject = new ConnectionController(this.host);
+            this.activeSolutionTracker.SimulateActiveSolutionChanged(true);
```
Comment (duncanp-sonar) [inline src/Integration.UnitTests/Connection/ConnectControllerTests.cs]: Are there any tests that check the connection status when the solution is not loaded?

File: src/Integration/Integration.csproj (hunk 3 above)
Comment (Evangelink) [reply src/Integration/Integration.csproj]: It does (haven't done any manual change).

File: src/Integration.UnitTests/Connection/ConnectControllerTests.cs (hunk 4 above)
Comment (Evangelink) [reply src/Integration.UnitTests/Connection/ConnectControllerTests.cs]: I have simply added tests for the new case so I can't tell you whether the other was already covered or not.

Comment (duncanp-sonar) [reply src/Integration.UnitTests/Connection/ConnectControllerTests.cs]: I meant tests simulating the event with _IsSolutionOpen = false_ , but on looking again I see you've added a default parameter to _SimulateActiveSolutionChanged_ so I guess there are tests that cover it.

**Comment by duncanp-sonar** [comment]
Closed: changes incorporated as part of  https://github.com/SonarSource/sonarlint-visualstudio/pull/412

**Pull request closed without merging**